src/ripple/app/misc/impl/AccountTxPaging.cpp

```diff
@@ -44,12 +44,19 @@ convertBlobsToTxResult(
 
     auto tr = std::make_shared<Transaction>(txn, reason, app);
 
-    tr->setStatus(Transaction::sqlTransactionStatus(status));
-    tr->setLedger(ledger_index);
-
     auto metaset =
         std::make_shared<TxMeta>(tr->getID(), tr->getLedger(), rawMeta);
 
+    // if properly formed meta is available we can use it to generate ctid
+    if (metaset->getAsObject().isFieldPresent(sfTransactionIndex))
```

Comment:
@scottschurr Can you look at this? Wouldn't sfTransactionIndex always exist? If I do `account_tx` on a non validated ledger I get "Ledger not validated." `SetStatus` values are optional. So can we just call this every time?

Reply:
Or how do I test `account_tx` with a tx that doesnt have a sfTransactionIndex?

Reply:
@dangell7, thanks for asking.  I can't immediately think of a situation where you could retrieve metadata for a transaction from a ledger and not have `sfTransactionIndex` filled in.  However I prefer the conservative approach of verifying that the field is present before accessing it.  Bugs happen.  Doing checks like this can prevent a bug elsewhere from turning catastrophic.

Taking a step back, we ran for a few years without code coverage as part of CI. And, thanks to some great work by good folks, we started getting code coverage reports again recently.  But we're still learning how to best apply the tool.  There will be code that can't be exercised by reasonable tests (testing `asserts` anyone?).  We have a new microscope and we've discovered that there are tiny mites living on our skin.  Over time we'll learn to relax and know we don't have to check that each individual mite is harmless.

Having said that, if I were bound and determined to exercise the `else` condition here, I'd doing it by calling the `convertBlobsToTxResult()` function directly.  That would allow me to pass in whatever I wanted in the `rawMeta` argument.  This test would pass in `rawMeta` that omitted an `sfTransactionIndex`.  I'd guess that getting that test in place might take a day or so of effort.

Reply:
Awesome. Thank you for this response. I do like the code coverage stuff. I will take a look at how to get some more coverage on this. `convertBlobsToTxResult` sounds pretty simple. Thanks